spec/models/firmware_spec.rb

```diff
@@ -0,0 +1,12 @@
+RSpec.describe Firmware do
+  describe "hardware" do
+    it "can set the hardware relationship" do
+      hardware = FactoryGirl.create(:hardware)
+      firmware = FactoryGirl.create(:firmware, :hardware => hardware)
+
+      expect(firmware.hardware).to eq(hardware)
+      expect(firmware.resource_id).to eq(hardware.id)
```

Comment:
Not generally into writing tests like this that seem to be testing the framework, but since we depend on this elsewhere I thought this was OK ;)

Reply:
@imtayadeway   LGTM.  Verified that association works in both directions with this change and a solid provider instance.